zephyr/lib/cpu.c

```diff
@@ -80,6 +89,23 @@ void cpu_notify_state_entry(enum pm_state state)
 			k_panic();
 		}
 
+		/* Suspend all DAI components before entering D3 state. */
+		list_for_item(clist, &ipc_get()->comp_list) {
+			icd = container_of(clist, struct ipc_comp_dev, list);
+			if (icd->type != COMP_TYPE_COMPONENT)
+				continue;
+
+			if (dev_comp_type(icd->cd) == SOF_COMP_DAI) {
+				mod = comp_mod(icd->cd);
+				cd = module_get_private_data(mod);
+				dd = cd->dd[0];
+				if (dai_remove(dd->dai->dev) < 0) {
```

Comment:
Hmm, @tmleman can you refresh why this is needed on SOF side? I mean Zephyr has a PM framework for devices and actually most of the DAI drivers we have, register to it. E.g. Intel dmic driver implements dmic_pm_action(). So Zephyr should already put the devices to the suspend -- not sure why we need this on SOF side.

Reply:
SOF manages DAI power states explicitly due to the requirement that all audio pipelines must be paused before entering D3. Zephyr's PM framework doesn't automatically suspend devices like DAIs during system PM transitions, as it's designed for devices to be suspended based on runtime usage. SOF's power management needs to synchronize DAI states with the DSP core during audio-specific power transitions, which is beyond the scope of Zephyr's device-level PM.

Reply:
@tmleman  please capture this in some comments inside the code so people can understand why can't we just use (yet) the Zephyr PM.

Reply:
Thanks @tmleman , and indeed I'll echo @dbaluta 's comment that please add a comment about this. This is different from e.g. Linux kernel (devices are suspended when system is suspended), so this can be surprising to many people. And looking at Zephyr docs, PM_DEVICE_SYSTEM_MANAGED states ""This option enables the system-managed device power management.  The power management subsystem will suspend
devices before entering a low power state. "" Isn't this what we want for SOF as well (and this option is set for SOF)? @ceolin @nashif can you chime in on this as well?

Reply:
Thanks @tmleman . I now realize I've been confused by the probe/remove functions. In Linux kernel, these are part of the device model (linux/device/driver.h) and they don't imply power up/down like used here. But in Zephyr, probe/remove are actually DAI specific methods, and the documentation explicitly mentions power-up/down. 
Now we don't need to have alignment, but as Zephyr is using similar naming (for device model and runtime-pm), this is somewhat unfortunate reuse of terms. But oh well, now that I realize that, then I understand the patch and am good with it.